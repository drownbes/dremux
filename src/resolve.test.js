const { Graph } = require('./graph');
const { graphResolve } = require('./resolve');

const logStr = [];

const simulateTask = (id, deps) => new Promise((resolve,reject) => {
  setTimeout(() => {
    if(deps) {
      logStr.push(`resolved ${id} with deps: ${[...deps.keys()]}`);
    } else {
      logStr.push(`resolved ${id}`);
    }

    resolve(`resolved ${id}`);
  }, 0);
});

describe('resolve', () => {
  test('graphResolve', async () => {
    const g = new Graph();
    g.addEdge(0, 3);
    g.addEdge(0, 6);
    g.addEdge(1, 3);
    g.addEdge(2, 3);
    g.addEdge(2, 4);
    g.addEdge(3, 5);


    await graphResolve(g, simulateTask).then(r => {
      logStr.push(r);
      expect(logStr).toMatchSnapshot();
    })

  });


})