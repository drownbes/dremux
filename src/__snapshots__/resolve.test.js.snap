// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolve graphResolve 1`] = `
Array [
  "resolved 6",
  "resolved 4",
  "resolved 5",
  "resolved 3 with deps: 5",
  "resolved 1 with deps: 3",
  "resolved 2 with deps: 3,4",
  "resolved 0 with deps: 3,6",
  Array [
    "resolved 6",
    "resolved 4",
    "resolved 5",
    "resolved 3",
    "resolved 1",
    "resolved 2",
    "resolved 0",
  ],
]
`;
